sdk/core/core-http/src/url.ts

```diff
@@ -128,12 +122,6 @@ export class URLQuery {
             }
             break;
 
-          case "Invalid":
-            if (currentCharacter === "&") {
-              currentState = "ParameterName";
```

Comment:
What was this branch used for?

Reply:
After it hit one '=' in parameter value it would skip characters until it sees '&' which starts a new parameter name/value pair

Reply:
Huh. This must have been motivated by a specific case. I wonder what it was.

Reply:
So current behavior is if parameter value contains '=' we would not add it into the dictionary.  I don't know whether there is a scenario that would use this behavior.

Reply:
/cc @daschult to see if he can remember why we treat '=' in parameter values as invalid?